packages/camera/camera_avfoundation/ios/Classes/FLTCam.m

```diff
@@ -1239,3 +1299,75 @@ - (void)setUpCaptureSessionForAudio {
   }
 }
 @end
+
+@implementation FLTCamMediaSettingsProvider
+
+#define AssertPositiveNumberOrNil(param)                                                        \
+  if (param != nil) {                                                                           \
+    NSAssert([param isKindOfClass:NSNumber.class], @"%@ is not a number: %@", @ #param, param); \
+    NSAssert(!isnan([param doubleValue]), @"%@ is NaN", @ #param);                              \
+    NSAssert([param doubleValue] > 0, @"%@ is not positive: %@", @ #param, param);              \
+  }
+
+- (instancetype)initWithFps:(nullable NSNumber *)fps
+               videoBitrate:(nullable NSNumber *)videoBitrate
+               audioBitrate:(nullable NSNumber *)audioBitrate
+                enableAudio:(BOOL)enableAudio {
+  AssertPositiveNumberOrNil(fps);
+  AssertPositiveNumberOrNil(videoBitrate);
+  AssertPositiveNumberOrNil(audioBitrate);
+
+  _fps = fps;
+  _videoBitrate = videoBitrate;
+  _audioBitrate = audioBitrate;
+  _enableAudio = enableAudio;
+
+  return self;
+}
+
+- (BOOL)lockDevice:(AVCaptureDevice *)captureDevice error:(NSError **)outError {
+  return [captureDevice lockForConfiguration:outError];
+}
+
+- (void)unlockDevice:(AVCaptureDevice *)captureDevice {
+  return [captureDevice unlockForConfiguration];
+}
+
+- (void)beginConfigurationForSession:(AVCaptureSession *)videoCaptureSession {
+  [videoCaptureSession beginConfiguration];
+}
+
+- (void)commitConfigurationForSession:(AVCaptureSession *)videoCaptureSession {
+  [videoCaptureSession commitConfiguration];
+}
+
+- (void)setMinFrameDuration:(CMTime)duration onDevice:(AVCaptureDevice *)captureDevice {
+  captureDevice.activeVideoMinFrameDuration = duration;
+}
+
+- (void)setMaxFrameDuration:(CMTime)duration onDevice:(AVCaptureDevice *)captureDevice {
+  captureDevice.activeVideoMaxFrameDuration = duration;
+}
+
+- (AVAssetWriterInput *)assetWriterAudioInputWithOutputSettings:
+    (nullable NSDictionary<NSString *, id> *)outputSettings {
+  return [AVAssetWriterInput assetWriterInputWithMediaType:AVMediaTypeAudio
+                                            outputSettings:outputSettings];
+}
+
+- (AVAssetWriterInput *)assetWriterVideoInputWithOutputSettings:
+    (nullable NSDictionary<NSString *, id> *)outputSettings {
+  return [AVAssetWriterInput assetWriterInputWithMediaType:AVMediaTypeVideo
+                                            outputSettings:outputSettings];
+}
+
+- (void)addInput:(AVAssetWriterInput *)writerInput toAssetWriter:(AVAssetWriter *)writer {
+  [writer addInput:writerInput];
+}
+
+- (NSMutableDictionary *)recommendedVideoSettingsForOutput:(AVCaptureVideoDataOutput *)output {
+  return [[output recommendedVideoSettingsForAssetWriterWithOutputFileType:AVFileTypeMPEG4]
+      mutableCopy];
```

Comment:
Similarly, why is the mutable copy part of the wrapper if that's not what the underlying API returns?

Reply:
implementation logic removed. including mutable copy part.